### ISSUE TITLE
Dig resolver incorrect result

### ISSUE THREAD

**Issue by divine** [remotelyliving/php-dns#36]
**Describe the bug**

Dig resolver doesn't correctly return result.

**To Reproduce**

```php
$dig = new Dig(new SpatieDNS(), new DigMapper(), Hostname::createFromString('ns1.yandex.ru'));
$result = $dig->getRecords('yandex.ru',  DNSRecordType::TYPE_A);
```

**$digResponse** data is:
```php
yandex.ru.\t\t300 IN A 77.88.55.55\r\n
yandex.ru.\t\t300 IN A 77.88.55.60\r\n
yandex.ru.\t\t300 IN A 5.255.255.55\r\n
yandex.ru.\t\t300 IN A 5.255.255.60\r\n
```
**parseDigResponseToRows**(string $digResponse) return $rows is:

```
array:1 [
  0 => array:5 [
    0 => "yandex.ru.300"
    1 => "IN"
    2 => "A"
    3 => "5.255.255.60yandex.ru.300"
    4 => "IN A 77.88.55.60yandex.ru.300 IN A 5.255.255.55yandex.ru.300 IN A 77.88.55.55"
  ]
]
```

**Expected behavior**
Should return correct results.

Thanks!

**Comment by remotelyliving** [comment]
Yikes! That's no good. I think maybe my crappy regex for parsing doesn't account for tabs vs spaces. Let me check on that and get a patch going.

**Comment by remotelyliving** [comment]
@divine hopefully fixed in https://github.com/remotelyliving/php-dns/tree/4.1.1

Thanks again, and let me know if it gives you any more issues

**Comment by divine** [comment]
@remotelyliving thank you so much for a great library and for the fast fix too!

It's working as expected now and I'm really happy 😄!

Thanks!

**Comment by remotelyliving** [comment]
@divine wooo! I'm glad you're finding it useful, and thanks again for the report 😸

**Comment by divine** [comment]
@remotelyliving Houston, we have a problem 😅.

Sorry, my bad, should have checked every response type, **TYPE_NS** and **TYPE_TXT** has some weird responses as well.

```php
$dig = new Dig(new SpatieDNS(), new DigMapper(), Hostname::createFromString('ns1.google.com'));
$result = $dig->getRecords('google.com',  DNSRecordType::TYPE_NS);
```
**$result** is:

```php
array:1 [
  0 => RemotelyLiving\PHPDNS\Entities\DNSRecord {#960
    -recordType: RemotelyLiving\PHPDNS\Entities\DNSRecordType {#956
      -type: "NS"
    }
    -hostname: RemotelyLiving\PHPDNS\Entities\Hostname {#957
      -hostname: "google.com."
    }
    -TTL: 345600
    -IPAddress: null
    -class: "IN"
    -data: RemotelyLiving\PHPDNS\Entities\NSData {#958
      -target: RemotelyLiving\PHPDNS\Entities\Hostname {#959
        -hostname: "ns3.google.com.google.com. 345600 in ns ns1.google.com.google.com. 345600 in ns ns4.google.com.google.com. 345600 in ns ns2.google.com."
      }
    }
  }
]

```
**$digResponse** data is:
```php
google.com.\t\t345600 IN NS ns3.google.com.\r\n
google.com.\t\t345600 IN NS ns1.google.com.\r\n
google.com.\t\t345600 IN NS ns2.google.com.\r\n
google.com.\t\t345600 IN NS ns4.google.com.\r\n
```

**parseDigResponseToRows**(string $digResponse) return **$rows** is:
```php
array:1 [
  0 => array:5 [
    0 => "google.com."
    1 => "345600"
    2 => "IN"
    3 => "NS"
    4 => "ns3.google.com.google.com. 345600 IN NS ns1.google.com.google.com. 345600 IN NS ns4.google.com.google.com. 345600 IN NS ns2.google.com."
  ]
]

```

____


For **TYPE_TXT**:

```php
$dig = new Dig(new SpatieDNS(), new DigMapper(), Hostname::createFromString('ns1.google.com'));
$result = $dig->getRecords('google.com',  DNSRecordType::TYPE_TXT);
```
**$result** is:

```php
array:1 [
  0 => RemotelyLiving\PHPDNS\Entities\DNSRecord {#959
    -recordType: RemotelyLiving\PHPDNS\Entities\DNSRecordType {#956
      -type: "TXT"
    }
    -hostname: RemotelyLiving\PHPDNS\Entities\Hostname {#957
      -hostname: "google.com."
    }
    -TTL: 3600
    -IPAddress: null
    -class: "IN"
    -data: RemotelyLiving\PHPDNS\Entities\TXTData {#958
      -value: "globalsign-smime-dv=CDYX+XFHUw2wml6/Gb8+59BsH31KzUr6c1l2BPvqKX8="google.com. 3600 IN TXT "v=spf1 include:_spf.google.com ~all"google.com. 3600 IN TXT "facebook-domain-verification=22rm551cu4k0ab0bxsw536tlds4h95"google.com. 300 IN TXT "docusign=05958488-4752-4ef2-95eb-aa7ba8a3bd0e"google.com. 300
IN TXT "docusign=1b0a6754-49b1-4db5-8540-d2c12664b289"
    }
  }
]


```
**$digResponse** data is:
```php
google.com.\t\t3600 IN\tTXT "v=spf1 include:_spf.google.com ~all"\r\n
google.com.\t\t3600 IN\tTXT "facebook-domain-verification=22rm551cu4k0ab0bxsw536tlds4h95"\r\n
google.com.\t\t300 IN TXT "docusign=1b0a6754-49b1-4db5-8540-d2c12664b289"\r\n
google.com.\t\t3600 IN\tTXT "globalsign-smime-dv=CDYX+XFHUw2wml6/Gb8+59BsH31KzUr6c1l2BPvqKX8="\r\n
google.com.\t\t300 IN TXT "docusign=05958488-4752-4ef2-95eb-aa7ba8a3bd0e"\r\n
```

**parseDigResponseToRows** (string $digResponse) return **$rows** is:
```php
array:1 [
  0 => array:5 [
    0 => "google.com."
    1 => "300"
    2 => "IN"
    3 => "TXT"
    4 => ""docusign=1b0a6754-49b1-4db5-8540-d2c12664b289"google.com. 3600 IN TXT "globalsign-smime-dv=CDYX+XFHUw2wml6/Gb8+59BsH31KzUr6c1l2BPvqKX8="google.com. 300 IN TXT "docusign=05958488-4752-4ef2-95eb-aa7ba8a3bd0e"google.com. 3600 IN TXT "v=spf1 include:_spf.google.com ~all"google.com. 3600 IN TXT "facebook-
domain-verification=22rm551cu4k0ab0bxsw536tlds4h95""
  ]
]


```

Thank you so much again!

**Comment by remotelyliving** [comment]
👀

**Comment by remotelyliving** [comment]
This is weird! I can't reproduce like I could the last time. What version of PHP and what OS are you on? Might be some silly differences in the regex engine that I'm just not picking up on.

**Comment by divine** [comment]
> This is weird! I can't reproduce like I could the last time. What version of PHP and what OS are you on? Might be some silly differences in the regex engine that I'm just not picking up on.

PHP version 7.4, OS is a Windows.  No problems on Linux.

The exact problem is happening after this regex line https://github.com/remotelyliving/php-dns/blob/b8647e6b481b8606b4eb119447c76c13ea7c529a/src/Resolvers/Dig.php#L88

Before:

```php
google.com. 345600 IN NS ns1.google.com.\r\n
google.com. 345600 IN NS ns2.google.com.\r\n
google.com. 345600 IN NS ns4.google.com.\r\n
google.com. 345600 IN NS ns3.google.com.
```

After that regex line this becomes:

`"google.com. 345600 IN NS ns1.google.com.google.com. 345600 IN NS ns3.google.com.google.com. 345600 IN NS ns2.google.com.google.com. 345600 IN NS ns4.google.com."`

on Linux it's returning correctly:

```php
google.com. 345600 IN NS ns1.google.com.
google.com. 345600 IN NS ns3.google.com.
google.com. 345600 IN NS ns2.google.com.
google.com. 345600 IN NS ns4.google.com.
```

Basically the problem is definitely with regex:

```
\r\n - on a windows computer
\r - on an Apple computer
\n - on Linux
```

Basically lines needs to be normalized between different OS, something like this?

```php
$normalized = preg_replace('~\r\n?~', "\n", (string) $tabsRemoved);
$breaksRemoved = preg_replace('/\s\s/m', '', (string) $normalized); 
```

Thanks!

**Comment by divine** [comment]
Hello,

After much testing, I think dig resolver isn't the best decision to use at all, it always returns different errors thus failing on each query.

However this might be a better solution https://github.com/reactphp/dns just in case.

Feel free to close this issue and again thank you for the library!

Thanks!

**Comment by remotelyliving** [comment]
Fixed in https://github.com/remotelyliving/php-dns/pull/48
